TLS Feature

ashish-vmware opened the pull request:
- Adding TLS support and certificate validations in current MQTT implementation
- Adding identity layer to encapsulate gateway identity and TLS configurations

Commit:
TLS Feature:
- Adding TLS support and certificate validations in current MQTT implementation
- Adding identity layer to encapsulate gateway identity and TLS configurations

diwanc commented on liota/transports/mqtt.py:
```diff
@@ -48,25 +49,78 @@ def on_message(self, client, userdata, msg):
+                # 1. Both Client certificate and key file should be present
+                # 2. If both are not there proceed without client certificate and key
+                # 3. If one of them is not there throw an error to notify one of them is missing
+            if client_cert_available != client_key_available:
```
1) Boolean operator "not" preferable over "!="
2) Also, think of splitting the first case into two; a) "Client cert file found. But key not found." b) "Client key file found. But cert file not found."

diwanc commented on liota/transports/mqtt.py:
```diff
@@ -48,25 +49,78 @@ def on_message(self, client, userdata, msg):
 
+        # Connect with MQTT Broker
+        self.client.on_connect = self.on_connect
+        self.client.connect(host=self.url, port=self.port, keepalive=60)
```
Is keepalive not a param to be collected at method level? If yes, lets do that and have a default value.

Commit:
Updates in TLS implementations
- Making keepalive parameter configurable
- Updating client certification validation method
- Adding sample MQTT configurations

Commit:
- Changes in certificate validations

diwanc commented on liota/transports/mqtt.py:
```diff
@@ -48,25 +49,82 @@ def on_message(self, client, userdata, msg):
+                # 1. Both Client certificate and key file should be present
+                # 2. If both are not there proceed without client certificate and key
+                # 3. If client certificate is not there throw an error
+                # 4. If client key is not there throw an error
```
This is more elegant.
